crypto/pkcs11key/key.go

```diff
@@ -166,12 +204,18 @@ func (ps *Key) setup(privateKeyLabel string) (err error) {
 	}
 	ps.privateKeyHandle = privateKeyHandle
 
-	publicKey, err := getPublicKey(ps.module, session, privateKeyHandle)
+	publicKey, err := ps.getPublicKey(ps.module, session, privateKeyHandle)
 	if err != nil {
 		ps.module.CloseSession(session)
 		return
 	}
-	ps.publicKey = publicKey
+	if ps.keyType == EC {
+		t := publicKey.(ecdsa.PublicKey)
```

Comment:
Good point. In theory we could do that with a switch statement like `switch pub := pub.(type) {` switch statement, but I don't feel strongly about that.

I do still think the change to emit pointers rather than values directly from `getPublicKey` is worthwhile, along with putting the `keyType` decision inside `getPublicKey`. What about this as a method signature:

`publicKey, keyType, err := ps.getPublicKey(...)`